src/lib/temple/back/main.ts

```diff
@@ -257,34 +259,52 @@ const getCurrentAccountPkh = async (): Promise<string | undefined> => {
   return frontState.accounts[0]?.publicKeyHash;
 };
 
-browser.runtime.onMessage.addListener(msg => {
-  switch (msg?.type) {
-    case ContentScriptType.ExternalLinksActivity:
-      const trackedCashbackServiceDomain = getTrackedCashbackServiceDomain(msg.url);
+const getAnalyticsUserId = async (): Promise<string | undefined> => {
+  const { [ANALYTICS_USER_ID_STORAGE_KEY]: userId } = await browser.storage.local.get(ANALYTICS_USER_ID_STORAGE_KEY);
 
-      if (trackedCashbackServiceDomain) {
-        Analytics.client.track('External Cashback Links Activity', { domain: trackedCashbackServiceDomain });
-      }
+  return userId;
+};
 
-      const trackedUrl = getTrackedUrl(msg.url);
+browser.runtime.onMessage.addListener(async msg => {
+  try {
+    switch (msg?.type) {
+      case ContentScriptType.UpdateAdsRules:
+        await updateRulesStorage();
+        return;
+      case E2eMessageType.ResetRequest:
+        return clearAsyncStorages().then(() => ({ type: E2eMessageType.ResetResponse }));
+    }
 
-      if (trackedUrl) {
-        getCurrentAccountPkh()
-          .then(accountPkh => Analytics.client.track('External links activity', { url: trackedUrl, accountPkh }))
-          .catch(console.error);
-      }
+    const accountPkh = await getCurrentAccountPkh();
 
-      break;
-    case ContentScriptType.ExternalAdsActivity:
-      getCurrentAccountPkh()
-        .then(accountPkh => Analytics.client.track('External Ads Activity', { url: msg.url, accountPkh }))
-        .catch(console.error);
-      break;
-    case ContentScriptType.UpdateSliseAdsRules:
-      updateRulesStorage().catch(console.error);
-      break;
-    case E2eMessageType.ResetRequest:
-      return clearAsyncStorages().then(() => ({ type: E2eMessageType.ResetResponse }));
+    switch (msg?.type) {
+      case ContentScriptType.ExternalLinksActivity:
+        const trackedCashbackServiceDomain = getTrackedCashbackServiceDomain(msg.url);
+
+        if (trackedCashbackServiceDomain) {
+          await Analytics.client.track('External Cashback Links Activity', { domain: trackedCashbackServiceDomain });
+        }
+
+        const trackedUrl = getTrackedUrl(msg.url);
+
+        if (trackedUrl) {
+          await Analytics.client.track('External links activity', { url: trackedUrl, accountPkh });
+        }
+
+        break;
+      case ContentScriptType.ExternalAdsActivity:
+        const userId = await getAnalyticsUserId();
+        await Analytics.trackEvent({
```

Comment:
is it ok that this event was previously sent with `client.track` fn?

Reply:
It wasn't ok